include/arch/arm/aarch32/irq.h

```diff
@@ -235,6 +235,15 @@ extern void _isr_wrapper(void);
 
 #endif /* _ASMLANGUAGE */
 
+#ifdef CONFIG_ARMV6_M_ARMV8_M_BASELINE
+#define NVIC_NODEID DT_INST(0, arm_v6m_nvic)
+#elif CONFIG_ARMV7_M_ARMV8_M_MAINLINE
+#define NVIC_NODEID DT_INST(0, arm_v7m_nvic)
+#elif CONFIG_ARMV8_M_BASELINE || CONFIG_ARMV8_M_MAINLINE
+#define NVIC_NODEID DT_INST(0, arm_v8m_nvic)
+#endif
+#define NUM_IRQ_PRIO_BITS DT_PROP(NVIC_NODEID, arm_num_irq_priority_bits)
```

Comment:
General question about this PR. Think it's worth namespacing these with ARCH, so ARCH_IRQ_PRIO_BITS etc.?

Reply:
preference to utilize DT_INST/compat over DT_NODELABEL.

Reply:
I still don't understand what's wrong with node labels. This seems to be exactly what they're for, and it would save some code. But never mind ...

Reply:
Its stems from history of devicetree being focused on dtb usage and thus nodelabels would go away in the dtb.  So nodelabels haven't been assumed to be part of a binding as such from a devicetree point of view.  We're utilizing them in zephyr for the case when you need to know the specific instance of device (ie like the nrf HAL API requires currently) as well as pinmux.  However the usage of nodelabel should be kept to a minimum.

Reply:
Thanks for the context. I still think that Zephyr's usage of devicetree is so different from Linux's that it doesn't seem crazy for us to adopt our own conventions in this regard. There doesn't seem to be any harm in using node labels, at least not that I'm aware of. At any rate, this PR seems fine to me.